core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/nlp/sentiment/arrow/NltkSentimentOperator.java

```diff
@@ -327,5 +387,229 @@ private int getFreeLocalPort() throws IOException {
             s.close();
         }
     }
+
+    private org.apache.arrow.vector.types.pojo.Schema convertToArrowSchema(Schema texeraSchema) {
+        List<Field> arrowFields = new ArrayList<>();
+        for (Attribute a : texeraSchema.getAttributes()) {
+            String name = a.getName();
+            Field field = null;
+            switch (a.getType()) {
+                case INTEGER:
+                    field = Field.nullablePrimitive(name, new ArrowType.Int(32, true));
+                    break;
+                case DOUBLE:
+                    field = Field.nullablePrimitive(name, new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE));
+                    break;
+                case BOOLEAN:
+//                    field = Field.nullablePrimitive(name, new ArrowType.Bool());
+//                    break;
```

Comment:
maybe add a comment here, explain that the current BOOLEAN type is actually a string, so it fall-through to the string case. We might change this in the future.

Reply:
Comment Added.